Task Tracker what you will build page (#1349)
mongodb/docs-realm: source/tutorial/preview.txt

.. _tutorial-task-tracker-preview:

===================
What you will build
===================

.. default-domain:: mongodb

Task Tracker App
----------------

Regardless of which frontend tutorial you choose to follow, you'll make a collaborative 
task tracker app where users can keep track of tasks in solo and team projects. 

Users of your app will be able to:

- Register themselves with email and password.
- Sign in to their account with their email and password and sign out later.
- View a list of projects they are a member of.
- View, create, modify, and delete tasks in projects.
- View a list of team members in their project.
- Add and remove team members to their project.

The UI varies by frontend, but the functionality is consistent. 
Here is a walkthrough of the finished iOS Task Tracker.

.. figure:: /images/task-tracker-walkthrough.gif
    :alt: screencapture walkthrough of ios task tracker app
    :figwidth: 300px

> Realm Architecture

Reviewer: This section is great. I love how you broke down the diagrams to show a single user and multiple users. Nice work!

------------------

This diagram demonstrates how data coming from one user's interactions flows
through the different components of the app. 

- The red flow shows what happens when a user, Alice, signs in to the app. 

>   The frontend communicates to the backend, which handles Authentication.

Reviewer: No need to capitalize authentication here. But it might be a good spot to link to:
>``:term:`authentication provider` ``

Or
> ``:ref:`authentication <users-and-authentication>` ``

- The black flow follows the action of Alice updating a task. The frontend writes

>   the update to the local realm database that sits on Alice's device. Then if the 

Reviewer: As a general note, anywhere on this page where we say "realm" directly should be updated to use the appropriate source constant. You'll find the source constants in [snooty.toml](https://github.com/mongodb/docs-realm/blob/fc9bc0c962a9a7166e8e013431cfe83b19fdb9d1/snooty.toml#L62).

So for this one, it would be:
```suggestion
  the update to the local {+client-database+} that sits on Alice's device. Then if the 
```

During the build process, those source constants are replaced by their definitions as outlined in the snooty.toml file. This lets us easily update terms if there are changes to product/marketing.

  device has an internet connection, the data goes to the Realm backend
  and :ref:`Atlas Cluster <mongodb-data-access>`.

.. image:: /images/tutorial-1-user-dataflow-diagram.png
    :alt: diagram showing how user actions flow through the app structure.

This second diagram demonstrates how the actions of two separate users with 
separate accounts and devices might interact in the Task Tracker. 

- The red flow happens first, showing Alice adding Bob to her project. Once Alice's device
  connects to the internet, this update is received by the Realm App and 
  sent to Bob's device over the internet. At the end of the red flow, Bob has read 
  and write access to Alice's project. 

> - The blue flow is started by Bob adding a task to Alice's project and reverses 

Reviewer: We could tweak this a little so it doesn't use passive voice.
```suggestion
- The blue flow starts with Bob adding a task to Alice's project and reverses 
```

  through all the steps that the red flow went through, again requiring an internet 
  connection on both devices for the data to be consistent everywhere.

.. image:: /images/tutorial-2-user-dataflow-diagram.png
    :alt: diagram showing how 2 connected users' actions flow through the app structure.

.. tip::

>   Generally all changes made on a device are stored locally

Reviewer: This content is great, but could use a little wordsmithing. I sometimes use an app called [Hemingway](https://hemingwayapp.com/) to get some readability statistics on content and rework it. I reworked this tip a little with Hemingway, and incorporated some of our glossary terms, and came up with this as an alternative for this tip's text:
> :term:`{+client-database+}` stores changes made on the device, until it can sync with the :term:`{+app+}`. When the device has an active internet connection, it syncs changes with the backend. When an offline device connects to the internet, it uploads and downloads any changesets. The user can then see changes that other users made while they were offline. :ref:`Realm Sync  <sync>` handles this process.

  until it can connect to the internet and Sync with the backend. Once the 
  device comes online it will be able to see all of the changes other users made during 
  the time it was offline and its changes will be available to all other 
  devices when they are online. This is facilitated by :ref:`Realm Sync <sync>`. 

What We Cover in this Tutorial
------------------------------

In building the Task Tracker, you will take advantage of some of the 

> built-in functions available in Realm and the functionality of 

Reviewer: This paragraph might be a good place to link to functions, Sync, and Realm app.

.. _tutorial-task-tracker-preview:

===================
What You Will Build
===================

.. default-domain:: mongodb

Task Tracker App
----------------

When you complete the Task Tracker tutorial, you'll make a collaborative
task tracker app where users can keep track of tasks in solo and team projects. 

The UI may vary slightly depending on which frontend client you build. 
This walkthrough of the iOS Task Tracker demonstrates what users can do:

- Register or sign in with an email and password
- View a list of projects in which the user is a member
- Select a project to view its tasks
- Change task status to In Progress or Done
- Add new tasks
- Delete tasks
- Add team members to projects
- View team member projects
- View and manage tasks in team members' projects

> |

Author: I actually put it there on purpose because it looked a bit weird to me with no space between the bullets and the gif. but if you think its better I can delete it

> |

Author: <img width="432" alt="Screen Shot 2021-09-20 at 6 34 38 PM" src="https://user-images.githubusercontent.com/9171876/134084835-bf3c8d4c-7b97-4bf7-b8d7-4bcd80cdca84.png">
Here is what it looks like without the extra space, seemed a bit tight

> |

Author: <img width="802" alt="Screen Shot 2021-09-20 at 6 35 58 PM" src="https://user-images.githubusercontent.com/9171876/134085155-2a6f3bd3-98c1-45bb-8f31-ca908d31ae3e.png">

.. figure:: /images/task-tracker-walkthrough.gif
    :alt: screencapture walkthrough of ios task tracker app
    :figwidth: 300px

Realm Architecture
------------------

This diagram demonstrates how data coming from one user's interactions flows
through the different components of the app. 

- The red flow shows what happens when a user, Alice, signs in to the app. 
  The frontend communicates to the backend, which handles 
  :ref:`authentication <users-and-authentication>`.
- The black flow follows the action of Alice updating a task. The frontend writes
  the update to the local {+client-database+} that sits on Alice's device. Then if the 
  device has an internet connection, the data goes to the {+app+} backend
  and the :ref:`{+atlas-short+} cluster <mongodb-data-access>`.

.. image:: /images/tutorial-1-user-dataflow-diagram.png
    :alt: diagram showing how user actions flow through the app structure.

> 

Reviewer: Nit: we don't need two newlines here, so let's delete one.
